### PR TITLE
feat(datalog-console): add datalog-console

### PR DESCRIPTION
**Pull request by groundedSAGE**
## PR Description
Adds a way to signal that a Datalog DB is available to the Chrome extension. 
Adds a listener to receive a request for the database as a string and send it as a string to the Chrome extension. 

### Testing
- [ ] added relevant test coverage
- [ ] no tests needed

### Docs
- [ ] added relevant docs 
    - preview them at https://homebase.io/docs/homebase-react/{BRANCH_NAME}/overview
- [ ] updated relevant sections in the README.md
- [ ] updated relevant docstrings in index.d.ts
- [ ] no docs needed

### Typescript
- [ ] added or edited relevant Typescript type declarations
- [ ] no type declaration updates needed

## Merging
For maintainers.

To merge, select "Squash and Merge". Then:
  1. Make sure the top commit message follows [Angular Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines),
  2. Delete all other commit messages in the description, but keep any lines designating [co-authors](https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) so contributors will retain credit for their contributions.

### CODE REVIEW

File: src/homebase/datalog_console.cljs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(ns homebase.datalog-console
+  (:require [goog.object :as gobj]
+            [cljs.reader]))
+
+(js/document.documentElement.setAttribute "__datalog-inspect-remote-installed__" true)
```
Comment (smothers) [inline src/homebase/datalog_console.cljs:5]: ":datalog-console.remote/installed?"

File: src/homebase/datalog_console.cljs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+(defn init-datalog-console
```
Comment (smothers) [inline src/homebase/datalog_console.cljs:7]: rename to init!

File: src/homebase/react.cljs [hunk 3]
```diff
@@ -7,7 +7,10 @@
    [clojure.set]
    [homebase.js :as hbjs]
    [datascript.core :as d]
-   [datascript.impl.entity :as de]))
+   [datascript.impl.entity :as de]
+   [homebase.datalog-console :as console]))
```
Comment (smothers) [inline src/homebase/react.cljs:11]: :as datalog-console

File: src/homebase/react.cljs [hunk 4]
```diff
@@ -134,6 +137,7 @@
         conn (d/create-conn (if schema
                               (merge (hbjs/js->schema schema) base-schema)
                               base-schema))]
+    (console/init-datalog-console {:conn conn}) ; initialise listener for datalog admin console
```
Comment (smothers) [inline src/homebase/react.cljs:140]: I don't think the comment is needed. The naming is clear enough

File: src/homebase/datalog_console.cljs (hunk 1 above)
Comment (groundedSAGE) [reply src/homebase/datalog_console.cljs:5]: Seems that these special characters are not allowed as an attribute on html document. Changed it to "__datalog-console-remote-installed__"

**Review by smothers — APPROVED** [review]

File: src/homebase/datalog_console.cljs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+(ns homebase.datalog-console
+  (:require [goog.object :as gobj]
+            [cljs.reader]))
+
+(js/document.documentElement.setAttribute "__datalog-console-remote-installed__" true)
```
Comment (rubinovitz) [inline src/homebase/datalog_console.cljs:5]: @groundedSAGE it looks like this breaks Next.js support for HBR with the following error: 
```
error - ReferenceError: document is not defined
    at /Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:1491:479
    at /Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:6:22
    at Object.<anonymous> (/Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:10:3)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at eval (webpack-internal:///homebase-react:1:18)
Installing CLJS DevTools 1.0.2 and enabling features :formatters :hints :async
ReferenceError: document is not defined
    at /Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:1491:479
    at /Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:6:22
    at Object.<anonymous> (/Users/jb/repos/supabase-js/example/next-todo/node_modules/homebase-react/dist/js/homebase.react.js:10:3)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at eval (webpack-internal:///homebase-react:1:18)
CLJS DevTools: some custom formatters were not rendered.
https://github.com/binaryage/cljs-devtools/blob/master/docs/faq.md#why-some-custom-formatters-were-not-rendered
CLJS DevTools: some custom formatters were not rendered.
https://github.com/binaryage/cljs-devtools/blob/master/docs/faq.md#why-some-custom-formatters-were-not-rendered
```
Can you please do error catching here and make it a warning when it fails vs breaking compilation?

**Pull request merged**